### ISSUE TITLE
WHATWG URL not working

### ISSUE THREAD

**Issue by fraxken** [lukeed/httpie#15]
Hi,

The Type definition for your method take `string | Url` but you seem to have forgotten to transform the URL to a standard string. If we send an URL, it throw the following error:

```
TypeError: Cannot read property 'startsWith' of undefined
    at urlToOptions (internal/url.js:1260:28)
    at request (https.js:293:15)
    at Promise (F:\Code\orders\node_modules\httpie\dist\httpie.js:13:13)
    at new Promise (<anonymous>)
    at send (F:\Code\orders\node_modules\httpie\dist\httpie.js:6:9)
    at test (F:\Code\orders\test\test.js:48:19)
    at asPromise (F:\Code\orders\node_modules\japa\build\src\Callable\index.js:7:18)
    at Callable.Promise (F:\Code\orders\node_modules\japa\build\src\Callable\index.js:64:9)
    at new Promise (<anonymous>)
    at Object.Callable (F:\Code\orders\node_modules\japa\build\src\Callable\index.js:10:12)
```

Best Regards,
Thomas

**Comment by lukeed** [comment]
Hey, can you show me exactly what you're sending? Any URL objects are left as they are. Strings are parsed into URL object via `url.parse`.

**Comment by fraxken** [comment]
hi,

A Node.js WHATWG URL:
https://nodejs.org/api/url.html#url_the_whatwg_url_api

```js
const { get } = require("httpie");
async function main() {
    const { statusCode } = await get(new URL("https://www.google.fr/"));
    console.log(statusCode);
}
main().catch(console.error);
```

Best Regards,
Thomas

**Comment by lukeed** [comment]
Thanks, I knew what it was 😄 But you're right, I was handling it incorrectly. Turns out that when you merge a `URL` into an object, it merges its `"URLContext"` and not the formatted object you see when printing/reading it directly. TIL

```js
let foo = new URL("https://www.google.fr/");

console.log(foo);
/*
URL {
  href: 'https://www.google.fr/',
  origin: 'https://www.google.fr',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.google.fr',
  hostname: 'www.google.fr',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '' }
*/

console.log({ ...foo });
/*
{ [Symbol(context)]:
   URLContext {
     flags: 400,
     scheme: 'https:',
     username: '',
     password: '',
     host: 'www.google.fr',
     port: null,
     path: [ '' ],
     query: null,
     fragment: null },
  [Symbol(query)]: URLSearchParams {} }
*/
```

This is also a different behavior than the (legacy) `Url` class, which behaved as described.

**Comment by lukeed** [comment]
Released in `v1.1.1` – thank you 🙌
